src/Umbraco.Web.UI.Client/src/views/propertyeditors/colorpicker/colorpicker.html

```diff
@@ -8,7 +8,8 @@
     <umb-color-swatches colors="model.config.items"
                         selected-color="model.value.value"
                         size="m"
-                        use-label="model.useLabel">
+                        use-label="model.useLabel"
+                        on-select="onSelect(color)">
     </umb-color-swatches>

```

Comment:
@kjac in v8 the `selected-color` has been updated to `model.value` - not sure if this would change something here?
https://github.com/umbraco/Umbraco-CMS/blob/temp8/src/Umbraco.Web.UI.Client/src/views/propertyeditors/colorpicker/colorpicker.html#L10

Also maybe compare with these changes in v8 where the `on-select` has been updated, so when clicking color in icon picker it updated `model.color`.
https://github.com/umbraco/Umbraco-CMS/pull/4053/files

Reply:
@bjarnef to be honest the entire color picker controller should be rewritten for V8 if it hasn't already been (though from what you're pointing out here it sounds like it might have). It's littered with support for legacy formats and a general lack of underscore usage 😄

Reply:
@kjac since it doesn't update label might be because in v7 it used `model.value.value` instad of `model.value` which is an object and contains both `value` and `label` property?

```
<umb-color-swatches 
        colors="model.config.items"
        selected-color="model.value"
        size="m"
        use-label="model.useLabel">
</umb-color-swatches>
```

Reply:
@kjac as I remember in v7 `model.value` is an object, where it the the legacy format convert the `model.value` (string) to an `model.value` (object) with value and label, where both have value of the old `model.value` (color code).

Reply:
or maybe it was only from the prevalues it did this.
https://github.com/umbraco/Umbraco-CMS/blob/dev-v7/src/Umbraco.Web.UI.Client/src/views/propertyeditors/colorpicker/colorpicker.controller.js#L52-L72

Reply:
@bjarnef like I said... this controller is pretty much a mess. I've just mended it a bit and done some tidying in the process. I'm not entirely sure what you're asking me to do here, but if you have some concrete improvements (other than rewriting the whole thing), by all means let me know.